src/main/java/com/itproject/petshome/controller/DonatorController.java

```diff
@@ -19,15 +23,20 @@
 @AllArgsConstructor
 @RestController
 @Validated
-@RequestMapping("/api/v1/donator")
-
+@RequestMapping("/api/v1/user")
 public class DonatorController {
     UserService userService;
+    SessionService sessionService;
     @Operation(summary = "add donation")
-    @PostMapping("/")
-//    Require Front end finish the prototype and discuss what input to make
-//    and if we need any services created
-    public DonationDTO addDonation() {
-        return null;
+    @PostMapping("/donate")
+    public DonationDTO addDonation(@RequestBody DonationDTO donationDTO) {
```

Comment:
userId and DonationInput should be here(coz Donation has Id which client don't know)